modulemd/tests/test-modulemd-common.c

```diff
@@ -82,6 +82,84 @@ test_modulemd_load_file (void)
 }
 
 
+static void
+test_modulemd_load_string (void)
+{
+  const gchar *yaml_string = NULL;
+  g_autoptr (GError) error = NULL;
+  g_autoptr (ModulemdModuleIndex) idx = NULL;
+  g_autofree gchar *output = NULL;
+
+  /* This function is a wrapper around lower-level functions, so it should be
+   * okay to just test basic success and failure here.
+   */
+
+  /* Trivial modulemd */
+  g_clear_error (&error);
+  g_clear_object (&idx);
+  yaml_string =
+    "---\n"
+    "document: modulemd\n"
+    "version: 2\n"
+    "data:\n"
+    "  name: trivialname\n"
+    "  stream: trivialstream\n"
+    "  summary: Trivial Summary\n"
+    "  description: >-\n"
+    "    Trivial Description\n"
+    "  license:\n"
+    "    module: MIT\n"
+    "...\n";
+  idx = modulemd_load_string (yaml_string, &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (idx);
+
+  /* Make sure loaded index dumps to string cleanly */
+  g_clear_error (&error);
+  g_clear_pointer (&output, g_free);
+  output = modulemd_module_index_dump_to_string (idx, &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (output);
+
+
+  /* NULL string should raise an exception */
+  g_clear_error (&error);
+  g_clear_object (&idx);
+  modulemd_test_signal = 0;
+  signal (SIGTRAP, modulemd_test_signal_handler);
+  idx = modulemd_load_string (NULL, &error);
+  g_assert_cmpint (modulemd_test_signal, ==, SIGTRAP);
+  g_assert_null (idx);
+
+
+  // An empty string returns non-NULL but empty index. Is that right?
+  g_clear_error (&error);
+  g_clear_object (&idx);
+  yaml_string = "";
+  idx = modulemd_load_string (yaml_string, &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (idx);
+
+  // Outputting the empty index returned above returns a validation error...
```

Comment:
Yes, it returns `MMD_ERROR_VALIDATE` and the `error->message` should contain "Index contains no modules".

Reply:
OK. It just seems asymmetric that it's valid to load an empty string/file, but not valid to dump it back out.

Reply:
I'm just going to drop this test that attempts to output the empty index since it's existing behavior and really does nothing to test `modulemd_load_string()`.